aggregate_root/lib/aggregate_root.rb

```diff
@@ -4,6 +4,22 @@
 require 'aggregate_root/default_apply_strategy'
 
 module AggregateRoot
+  module ClassMethods
+    def on(*event_klasses, apply_strategy:  DefaultApplyStrategy.new, &block)
+      event_klasses.each do |event_klass|
+        handler_name = apply_strategy.handler_name_by_class(event_klass)
+        define_method handler_name do |event|
+          instance_exec event, &block
+        end
+        private handler_name
+      end
+    end
+  end
+
+  def self.included(host_class)
+    host_class.extend(ClassMethods)
```

Comment:
could we be explicit and ask the client to use:

```
class Order
  include AggregateRoot
  extend AggregateRoot::On
```

There might be a better name  then `AggregateRoot::On` maybe `AggregateRoot::OnHandlers` or `AggregateRoot::EasyHandlers` ?

Reply:
why? having it enabled by default making no difference, its up to developer to decide to use it or not

Reply:
> why?

less magic, more explicit. when you look into the class you can easily see where the additional class methods are coming from.

Reply:
then you will need as well ;)
```ruby
include Aggregate::Apply
```

Reply:
after all those are part of main functionality